USAGE.md

```diff
@@ -1,30 +1,144 @@
+    - [Build caching with private S3 buckets](#build-caching-with-private-s3-buckets)
+    - [Using GitHub OAuth to issue S3 credentials via Hashicorp Vault](#using-github-oauth-to-issue-s3-credentials-via-hashicorp-vault)
+
+## Quick start
+
+So, you have cloned a repo that you've heard has a devcontainer. You can see the file(s) for yourself by looking
+in your repo's `.devcontainer` folder. You may find a `devcontainer.json` file, or you may find some number of folders, such as `cuda12.0-conda` and `cuda12.0-pip`. If you find folders, these each contain a `devcontainer.json`
+file. These files specify how to create and run a container that leaves you with a good setup to do development.
+
+There are at least 2 ways to consume these devcontainer.json files. VS Code was
+the original home of devcontainers prior to becoming an open specification, and
+it remains a good way to use them. If you prefer not to use VS Code, the
+[devcontainers-cli](https://code.visualstudio.com/docs/devcontainers/devcontainer-cli)
+application will build the devcontainer and allow you to interact with it.
+
+### Using devcontainers in VS Code
+
+[The VS Code
+docs](https://code.visualstudio.com/docs/devcontainers/containers#_quick-start-open-an-existing-folder-in-a-container)
+are the definitive source of information for this topic.
+
+Specifically for RAPIDS repos, we frequently have multiple folders for different
+library configurations. Pay attention to which build environment you need when
+launching your devcontainer. You can switch between them
+
+### Using devcontainers from the terminal
+
+The [devcontainer-cli](https://code.visualstudio.com/docs/devcontainers/devcontainer-cli) project
+allows you to run and interact with devcontainers from the terminal. It uses NodeJS, so you need
+to install that in order to run it.
+
+You need to specify the devcontainer.json file and workspace folder to use when starting the devcontainer:
 
-* [Using the pre-built images](#using-the-pre-built-images)
-  * [Using in `devcontainer.json`](#using-in-devcontainerjson)
-* [Build caching with sccache](#build-caching-with-sccache)
+```
+devcontainer up --config .devcontainer/cuda12.0-pip/devcontainer.json --workspace-folder=.
+```
+
+## Available tools in the devcontainer
+
+### Generated build scripts
+
+Several scripts are generated for you based on the contents of manifest.yaml. By
+default, manifest.yaml comes from [the RAPIDS/devcontainers
+repo](https://github.com/rapidsai/devcontainers/blob/branch-24.02/features/src/rapids-build-utils/opt/rapids-build-utils/manifest.yaml),
+but you can use your own local manifest.yaml file. Refer to the "Adding
+projects" section.
+
+The generated scripts are:
+* /usr/bin/build-*
+* /usr/bin/clean-*
+* /usr/bin/clone-*
+* /usr/bin/configure-*
+
+Here the * is a placeholder for the project name and the kind of
+build. For example, a project with a python component in manifest.yaml
+will have `build-ucxx-python`.
```

Comment:
I'd avoid using UCXX as an example since it's new, somewhat different than other RAPIDS projects in its version scheme and build processes, and is a target to upstream (so things could change dramatically). Maybe use cuDF or RAFT as an example?